clang/test/dpct/helper_files_ref/include/util.hpp

```diff
@@ -524,6 +524,24 @@ class logical_group {
   }
 };
 }
+
+/// Get the number of bytes of free and total memory on the SYCL device.
+/// \param [out] the number of bytes of free memory on the SYCL device.
+/// \param [out] the number of bytes of total memory on the SYCL device.
+void get_sycl_mem_info(size_t &free_mem, size_t &total_mem) {
+  const auto &device = dpct::get_current_device();
+#if (defined(__SYCL_COMPILER_VERSION) && __SYCL_COMPILER_VERSION >= 20220927)
+  if (!device.has(sycl::aspect::ext_intel_free_memory)) {
+    throw std::runtime_error("ext_intel_free_memory is not supported.\n");
+  }
+  free_mem = device.get_info<sycl::ext::intel::info::device::free_memory>();
+#else
+  static_assert(
```

Comment:
Can we make it compiler warning and not error and return 0 for free memory, so that usage of other compilers is not blocked?

Reply:
Yes we can do that.